x/genutil/gentx.go

```diff
@@ -53,9 +54,8 @@ func ValidateAccountInGenesis(
 		func(bal bankexported.GenesisBalance) (stop bool) {
 			accAddress := bal.GetAddress()
 			accCoins := bal.GetCoins()
-
 			// ensure that account is in genesis
-			if accAddress.Equals(addr) {
+			if strings.Compare(accAddress, addr.String()) == 0 {
```

Comment:
shouldn't we use equalfold here instead of compare? Given that it reads something the user can input.